grid2op/tests/test_Opponent.py

```diff
@@ -126,5 +142,139 @@ def test_env_opp_attack_budget_ts(self):
                 assert env.opponent_budget_per_ts == 0.5
                 assert env.oppSpace.budget == 0.
 
+    def test_RandomLineOpponent_not_enough_budget(self):
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            init_budget = 3
+            with make("rte_case14_realistic",
+                      test=True,
+                      opponent_init_budget=init_budget,
+                      opponent_budget_per_ts=0.,
+                      opponent_action_class=TopologyAction,
+                      opponent_budget_class=BaseActionBudget,
+                      opponent_class=TestRandomLineOpponent) as env:
+                obs = env.reset()
+                assert env.oppSpace.budget == init_budget
+                # The opponent can attack
+                for i in range(init_budget):
+                    obs, reward, done, info = env.step(env.action_space())
+                    assert env.oppSpace.budget == init_budget - i - 1
+                    assert env.opponent_class.picked_attack != env.action_space({})
+                # There is no more budget
+                assert env.oppSpace.budget == 0
+                obs, reward, done, info = env.step(env.action_space())
+                assert env.opponent_class.picked_attack == env.action_space({})
+
+    def test_RandomLineOpponent_attackable_lines(self):
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            init_budget = 1000
+            tries = 30
+            attackable_lines_case14 = ["1_3_3", "1_4_4", "3_6_15", "9_10_12", "11_12_13", "12_13_14"]
+            with make("rte_case14_realistic",
+                      test=True,
+                      opponent_init_budget=init_budget,
+                      opponent_budget_per_ts=0.,
+                      opponent_action_class=TopologyAction,
+                      opponent_budget_class=BaseActionBudget,
+                      opponent_class=TestRandomLineOpponent) as env:
```

Comment:
seed not set, and check that it is always attacked in the same order, and also check that all powerlines in the list are attacked. (you might change the seed for that)

Reply:
done